test/DebugInfo/DebugInfoUnsupported.ll

```diff
@@ -0,0 +1,49 @@
+; This test checks that metadata with unknown DWARF operations is translated
+; to "empty" debug info to SPIR-V and back.
+; TODO: remove this test as soon as more DWARF operations are supported.
+
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv %t.spv -to-text -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV: [[#DEBUG_NONE:]] [[#]] DebugInfoNone
+; CHECK-SPIRV: [[#DEBUG_TYPE_FUNC:]] [[#]] DebugTypeFunction [[#]] [[#DEBUG_NONE]]
+; CHECK-SPIRV: [[#DEBUG_FUNC:]] [[#]] DebugFunction [[#]] [[#DEBUG_TYPE_FUNC]] [[#]] [[#]] [[#]] [[#]] [[#]] [[#]] [[#]] [[#]] [[#DEBUG_NONE]]
+; CHECK-SPIRV: [[#DEBUG_LOC_VAR:]] [[#]] DebugLocalVariable [[#]] [[#]] [[#]] [[#]] [[#]] [[#DEBUG_FUNC]]
+; CHECK-SPIRV: [[#EXPRESSION:]] [[#]] DebugExpression
+; CHECK-SPIRV: DebugValue [[#DEBUG_LOC_VAR]] [[#]] [[#EXPRESSION]]
+
+target datalayout = "e-m:o-i64:64-f80:128-n8:16:32:64-S128"
+target triple = "spir64-unknown-unknown"
+
+declare void @llvm.dbg.value(metadata, metadata, metadata) nounwind readnone speculatable
+
+define void @DbgIntrinsics() sanitize_memtag {
+entry:
+  %x = alloca i32, align 4
+; CHECK-LLVM-NOT: DW_OP_LLVM_arg
+; CHECK-LLVM: call void @llvm.dbg.value(metadata i32* %x, metadata ![[#]], metadata !DIExpression())
+; CHECK-LLVM-NOT: DW_OP_LLVM_arg
```

Comment:
Doesn't the first CHECK-LLVM-NOT: DW_OP_LLVM_arg scan all of the module ? I mean the second check seems to be not needed.

Reply:
No, the behavior you described works only when `--implicit-check-not` option is enabled in FileCheck. Otherwise, it works like that:

> The “CHECK-NOT:” directive is used to verify that a string doesn’t occur between two matches (or before the first match, or after the last match)

Although, it's possible to use the option and to replace two checks with one.

Reply:
No need to replace then, thanks